File: 1. Two Sum.cpp
// 1. Two Sum

#include <vector>
#include <unordered_map>
using namespace std;


class Solution {
    public:
        vector<int> twoSum(vector<int>& nums, int target) {
            unordered_map<int, int> map;

            int n = nums.size();
            for (int i = 0; i < n; i++) {
                if (map.find(target - nums[i]) != map.end()) {
                    return {map[target - nums[i]], i};
                } else {
                    map[nums[i]] = i;
                }
            }
            return {};
        }
    };

File: 1. Two Sum.py
# 1. Two Sum
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = defaultdict(int)
        for i in range(len(nums)):
            need = target - nums[i]
            if need in dic:
                return [dic[need], i]
            
            dic[nums[i]] = i
        